Server/ConcordiaCurriculumManager/Services/CourseGroupingService.cs

```diff
@@ -307,4 +308,35 @@ public async Task<CourseGrouping> PublishCourseGrouping(Guid commonIdentifier)
 
     private async Task<Dossier> GetDossierDetailsByIdOrThrow(Guid id) => await _dossierRepository.GetDossierByDossierId(id)
         ?? throw new NotFoundException("The dossier does not exist.");
+
+    public async Task<List<CourseGrouping?>> GetCourseGroupingsByDossierAndName(Guid dossierId, string searchQuery)
+    {
+        var dossier = await _dossierRepository.GetDossierByDossierId(dossierId);
+        if (dossier == null)
+        {
+            throw new NotFoundException($"Dossier with ID {dossierId} not found.");
+        }
+
+        var groupingsByName = await _courseGroupingRepository.GetCourseGroupingsLikeName(searchQuery);
+
+        var creationGroupingsInDossier = dossier.CourseGroupingRequests
+            .Where(req => req.RequestType == RequestType.CreationRequest && req.CourseGrouping != null && req.CourseGrouping.Name.Contains(searchQuery, StringComparison.OrdinalIgnoreCase))
+            .Select(req => req.CourseGrouping)
+            .ToList();
+
+        var combinedGroupings = groupingsByName.Concat(creationGroupingsInDossier)
+            .DistinctBy(g =>  g!.Id)
+            .ToList();
+
+        var deletionGroupingIds = dossier.CourseGroupingRequests
+            .Where(req => req.RequestType == RequestType.DeletionRequest)
+            .Select(req => req.CourseGroupingId)
+            .ToList();
```

Comment:
This endpoint doesn't quite filter out groupings with a deletion request from the output. I think it's because here you use the grouping IDs to filter, rather than using the CommonIdentifier of the grouping, which tracks the same grouping across different versions.